### PR TITLE
[TACHYON-497] Update documentation on deploy module

### PR DESCRIPTION
**Pull request by uronce-cc**
There are documentations on deploy module both in README.md and on tachyon-project.org, a lot of contents are overlapped, some details are mentioned in one place but not in another, that's confusing for a new user.  In this PR, only documentations relating to contributors of deploy module are kept in README.md. 

Also, there is a standalone page for each provider of the deploy module, but since the workflow is similar, that's unnecessary and hard to maintain. 

AWS related configurations are confusing for a new AWS user, this PR provides more explanation and reference URLs in ec2 configuration file. 

All explanations of configurations are put in the configuration files themselves instead of maintaining in docs.

### CODE REVIEW

**Comment by calvinjia** [comment]
Could you tag this with the JIRA ticket please (497)?

File: deploy/vagrant/conf/init.yml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 Memory: 1024 #MB
 # Total is the number of VMs to start
 Total: 2
 # Addresses are the internal IPs given to each VM.
 # The last one is designated as Tachyon master
 # For VirtualBox: the addresses can be arbitrary
-# For AWS: the addresses should be within the same availability zone, like "172.31.32.155"
+# For AWS: the addresses must be in the same subnet, to find available subnets:
+#   go to AWS VPC web console and choose the same region and availability zone 
+#   as you specify in ec2-config.yml, look into "Subnets" on left panel
```
Comment (gpang) [inline deploy/vagrant/conf/init.yml]: I think you renamed the file to ec2.yml.

File: docs/Deploy-Module.md [hunk 2]
```diff
@@ -1,39 +1,96 @@
...
+Then you need to set shell environment variables `AWS_ACCESS_KEY`
+and `AWS_SECRET_KEY`, refer to [this doc](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html). 
+
+Then download [key pair](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html).
+
+Install aws vagrant plugin. To date, 0.5.0 plugin is tested.
+
+    vagrant plugin install vagrant-aws
+
+Then update configurations in `conf/ec2-config.yml`.
```
Comment (gpang) [inline docs/Deploy-Module.md]: I think this was renamed to ec2.yml

File: docs/Deploy-Module.md [hunk 3]
```diff
@@ -1,39 +1,96 @@
...
+
+There are three kinds of configuration files under `conf`. All the configuration files use [yaml syntax](http://en.wikipedia.org/wiki/YAML).
+
+1. configuration of virtual machine itself in `init.yml`. 
+2. provider specific configuration, like `ec2.yml`, `openstack.yml`.
+3. software configuration, like `tachyon.yml`, `spark.yml`, `ufs.yml`(ufs means under filesystem). 
+
+Meanings of fields in the configurations are explained in detail in the yml files.
+
+## AWS EC2
```
Comment (gpang) [inline docs/Deploy-Module.md]: Is there any info on the ec2-cli commandline tools? I think that should be somewhere...

**Comment by uronce-cc** [comment]
@gpang thx for verification.

**Comment by gpang** [comment]
It looks like you got rid of a lot of documentation, but I don't see where they were added somewhere else. Does this mean we will have less documentation than before?

**Comment by uronce-cc** [comment]
@gpang before, each target deployment platform has a separate documentation, but the only difference should be configuration, the overall workflow is the same. So I only describe the workflow in one page, and put configuration explanations to the configuration files instead of documentation.

File: deploy/vagrant/conf/tachyon.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# Github means cloning and compiling github repo
+Type: Local
+
+# github repo and version(can be branch, commit hash)
+Github:
+  Repo: https://github.com/amplab/tachyon
+  Version: branch-0.5
+
+Release:
+  Dist: tachyon-0.5.0-bin.tar.gz # name of release tarball in https://github.com/amplab/tachyon/releases
```
Comment (gpang) [inline deploy/vagrant/conf/tachyon.yml]: Should it default to a more recent version?

File: docs/Deploy-Module.md [hunk 5]
```diff
@@ -1,39 +1,98 @@
...
+
+1. `cd TACHYON/deploy/vagrant`, replace TACHYON with the Tachyon repo on your host, you can clone the latest from [github](https://github.com/amplab/tachyon.git)
+2. `./run_vb.sh`, it will be slow when you first run this command, because vagrant will download a centos6.5 virtual machine, and install necessary software in this machine. The console output is verbose enough for you to know what's going on. 
+3. After the above command, a purple line in the output tells you the IP of TachyonMaster, the IP should be "192.168.1.12", visit `http://192.168.1.12:19999` in your browser, you should see Tachyon's web UI
+4. Visit `http://192.168.1.12:50070` in your browser, you should see Hadoop's web UI!
+5. `vagrant ssh TachyonMaster`, and you should have entered the virtual machine named TachyonMaster
+6. `/tachyon/bin/tachyon runTests` to run some tests against Tachyon
+7. After the tests all pass, visit Tachyon web UI: `http://192.168.1.12:19999` again, in `Browse FileSystem`, you should see files written to Tachyon by the tests run above
+8. If you don't want to play around in the cluster any more, `vagrant destory` to delete the virtual machines.
```
Comment (gpang) [inline docs/Deploy-Module.md]: 'destory' -> 'destroy'

**Comment by gpang** [comment]
LGTM.

Thanks for the improved docs, @uronce-cc !

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**